draft-ietf-dnsop-svcb-https.md

```diff
@@ -1228,12 +1228,18 @@ Section 12.1 of HSTS}}.
 
 ## Use of HTTPS RRs in other protocols
 
-All protocols employing "http://" or "https://" URLs SHOULD respect HTTPS RRs.
-For example, clients that
+All HTTP connections to named origins are eligible to use HTTPS RRs, even
+when HTTP is used as part of another protocol or without an explicit HTTP
+URL.  For example, clients that
 support HTTPS RRs and implement the altered WebSocket {{!WebSocket=RFC6455}}
 opening handshake from the W3C Fetch specification {{FETCH}} SHOULD use HTTPS RRs
 for the `requestURL`.
 
+When HTTP is used in a context where URLs or redirects are not applicable
+(e.g. Proxy Auto-Configuration), clients MUST implement transport security
+equivalent to the HSTS behavior specified in {{hsts}} or forego the use of
+HTTPS records.
```

Comment:
What does it mean to forego the use of HTTPS records, since the only use of HTTPS records in these context is the upgrade itself?

That is, in the context of an HTTPS proxy, the HSTS behavior is implemented by doing nothing, because you're already on HTTPS. This clause then never applies. In the context of an HTTP proxy, you may or may not implement the HSTS behavior... but since HTTPS records are, well, for HTTPS, there isn't any other use of HTTPS records to forego.

Reply:
Reworded